enhancements/sig-architecture/17-cluster-features/README.md

```diff
@@ -0,0 +1,110 @@
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+There are requirements about selecting `ManagedClusters` based on the certain features of the managed cluster by `Placement` API. These requirements include:
+
+1. Select a set of the clusters that a certain `ManagedClusterAddon` is enabled. For instance, a user wants to install an application that needs multicluster service connectivity, the use will need to select a set of clusters that `Submariner` addon is enabled by using `Placement` API.
+2. Select a set of the clusters with a certain capability. For instance, a user wants to deploy an AI workload to a set of the clusters in which GPUs are installed.
+
+## Motivation
+
+This proposal is to standardise the labels and `ClusterClaims` that are categorized as features of a `ManagedCluster`.  The idea is inspired by [node features](https://kubernetes-sigs.github.io/node-feature-discovery/) in kubernetes.
+
+### Goals
+
+- define the format of labels and `ClusterClaims` for `ManagedClusters` to represent a feature in managed cluster
+- implement addon as feature labels in `registration`.
+
+## Proposal
+
+### Design Details
+
+Features can be classified to 3 categories:
+
+1. Features enabled by hub cluster: these features are enabled by user on the hub cluster to a certain set of managedclusters. For example, when user enables an addon on a managed cluster, the cluster will have a feature relates to the addon. Each of these features can be represented as a label on the `ManagedCluster`.
+
+2. Existing node features on mananged cluster: these are features discovered by node feature discovery daemon in kubernetes cluster and has been represented as a label in the node of the managed cluster. Most of such features are physical device related. Each of these features should be represented as a `ManagedClusterClaim`.
+
+3. Existing cluster-wide features on managed cluster: these features resides in  managed cluster, but are not specific to a certain node. Examples includes whether a global load balancer service is supported in this managed cluster or a certain storage can be used with `PVC`. Each of these features should be represented as a `ManagedClusterClaim`.
+
+
+### Feature labels
+
+Feature labels should have the prefix of `feature.open-cluster-management.io`. Each addon will be represented as a feature label in the format of `feature.open-cluster-management.io/addon-<addon name>`. The values of the lable are `available`, `unhealthy`, `unreachable`. For example, the `application-manager` addon will be represented as a label on `ManagedCluster` as:
+
+```
+feature.open-cluster-management.io/addon-application-manager: available
+```
+
```

Comment:
1. I guess the label will be automatically removed if the addon is disabled.
2. What will the latency for the label to be present?
3. Can we have also the negative, for identity provider we are also looking to clusters which doesn't have a feature. Will that be possilbe, for example we want to know the managedcluster which doesn't have GRC.

Reply:
label select can have `notin`